semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -170,6 +166,7 @@ trait TextDocumentOps { self: SemanticdbOps =>
             if (gsym.isUselessSymbolInformation) return
             val symbol = gsym.toSemantic
             if (symbol == Symbols.None) return
+            if (config.symbols.isLocals && !symbol.isLocal) return
```

Comment:
Indeed, which makes `symbols:all` a good default for now. I'm eager to explore using this option however if it can shave off valuable overhead, getting the overhead to an absolute minimum will make it easier to convince people to flip the switch and have semanticdb enabled by default.